### PR TITLE
New tethys quotas

### PR DESCRIPTION
**Pull request by tylorbayer**

### CODE REVIEW

File: tethys_quotas/helpers.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
+"""
```
Comment (swainn) [inline tethys_quotas/helpers.py]: Rename file to utilities.py to match pattern set in other tethys django apps.

File: tethys_services/utilities.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 """
 ********************************************************************************
-* Name: utilities.py
```
Comment (swainn) [inline tethys_services/utilities.py]: Revert this change The file is still called utilities.py

File: docs/conf.py [hunk 3]
```diff
@@ -264,7 +265,7 @@ def __getattr__(cls, name):
     # 'pointsize': '10pt',
 
     # Additional stuff for the LaTeX preamble.
-    # 'preamble': '',
```
[the remark this answers is not in the document]
Comment (tylorbayer) [reply docs/conf.py]: ?

File: tethys_apps/utilities.py [hunk 4]
```diff
@@ -1,6 +1,6 @@
 """
 ********************************************************************************
-* Name: utilities.py
```
[the remark this answers is not in the document]
Comment (tylorbayer) [reply tethys_apps/utilities.py]: ?

[the remark this answers is not in the document]
Comment (tylorbayer) [reply tethys_apps/utilities.py]: I am not even sure that this change is suggesting

File: tethys_compute/utilities.py [hunk 5]
```diff
@@ -1,6 +1,6 @@
 """
 ********************************************************************************
-* Name: utilities.py
```
[the remark this answers is not in the document]
Comment (tylorbayer) [reply tethys_compute/utilities.py]: ? again.. not sure

File: tethys_services/utilities.py (hunk 2 above)
Comment (tylorbayer) [reply tethys_services/utilities.py]: ? same as others above

File: tethys_apps/utilities.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (tylorbayer) [reply tethys_apps/utilities.py]: tethys_apps has a helpers.py and utilities.py

File: tethys_compute/utilities.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (tylorbayer) [reply tethys_compute/utilities.py]: the name was just helpers in the header. i changed it though

File: tethys_apps/utilities.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (tylorbayer) [reply tethys_apps/utilities.py]: also renamed in back in the header

File: tethys_quotas/apps.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+class TethysQuotasConfig(AppConfig):
+    name = 'tethys_quotas'
+    verbose_name = 'Tethys Quotas'
+
+    def ready(self):
+        try:
+            sync_resource_quota_handlers()
+        except:  # noqa
+            log.warning("RQ table not created yet")
```
[the remark this answers is not in the document]
Comment (tylorbayer) [reply tethys_quotas/apps.py]: hmm.. this was already changed in my local dir... not sure how this didn't get updated

File: tethys_apps/base/workspace.py [hunk 7]
```diff
@@ -177,3 +187,259 @@ def remove(self, item):
...
             os.remove(full_path)
+
+    def get_size(self, units='b'):
+        units = units.lower()
+
+        total_size = 0
+        for file in self.files(True):
+            total_size += os.path.getsize(file)
+
+        if units == 'b':
```
[the remark this answers is not in the document]
Comment (tylorbayer) [reply tethys_apps/base/workspace.py]: I added a _get_storage_units function to quotas utilities and used it to get the conversion factor. I think its better. thanks

File: docs/conf.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (tylorbayer) [reply docs/conf.py]: I'm pretty sure that wasn't me. And I am not exactly sure what to revert.

File: tethys_portal/views/admin.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+@staff_member_required
+def clear_workspace(request, app_id):
+    """
+    Handle clear workspace requests.
+    """
+    url = '/admin/tethys_apps/tethysapp/{}/change/'.format(app_id)
+    app = TethysApp.objects.get(id=app_id)
+
+    apps_s = SingletonHarvester().apps
```
[the remark this answers is not in the document]
Comment (tylorbayer) [reply tethys_portal/views/admin.py]: I added a get_app_class function to tethys_apps utilities where you pass in the model and get the class, I also changed the get_active_app function to have an optional arg 'get_class' which calls the new function.
